catalog/resource_catalog.go

```diff
@@ -17,56 +18,16 @@ func ucDirectoryPathSuppressDiff(k, old, new string, d *schema.ResourceData) boo
 	return false
 }
 
-type CatalogsAPI struct {
-	client  *common.DatabricksClient
-	context context.Context
-}
-
-func NewCatalogsAPI(ctx context.Context, m any) CatalogsAPI {
-	return CatalogsAPI{m.(*common.DatabricksClient), context.WithValue(ctx, common.Api, common.API_2_1)}
-}
-
 type CatalogInfo struct {
-	Name         string            `json:"name" tf:"force_new"`
-	Comment      string            `json:"comment,omitempty"`
-	StorageRoot  string            `json:"storage_root,omitempty" tf:"force_new"`
-	ProviderName string            `json:"provider_name,omitempty" tf:"force_new,conflicts:storage_root"`
-	ShareName    string            `json:"share_name,omitempty" tf:"force_new,conflicts:storage_root"`
-	Properties   map[string]string `json:"properties,omitempty"`
-	Owner        string            `json:"owner,omitempty" tf:"computed"`
-	MetastoreID  string            `json:"metastore_id,omitempty" tf:"computed"`
-}
-
-type Catalogs struct {
-	Catalogs []CatalogInfo `json:"catalogs"`
-}
-
-func (a CatalogsAPI) createCatalog(ci *CatalogInfo) error {
-	return a.client.Post(a.context, "/unity-catalog/catalogs", ci, ci)
-}
-
-func (a CatalogsAPI) getCatalog(name string) (ci CatalogInfo, err error) {
-	err = a.client.Get(a.context, "/unity-catalog/catalogs/"+name, nil, &ci)
-	return
-}
-
-func (a CatalogsAPI) deleteCatalog(name string) error {
-	return a.client.Delete(a.context, "/unity-catalog/catalogs/"+name, nil)
-}
-
-func (a CatalogsAPI) forceDeleteCatalog(name string) error {
-	schemasAPI := NewSchemasAPI(a.context, a.client)
-	schemas, err := schemasAPI.listByCatalog(name)
-	if err != nil {
-		return err
-	}
-	for _, s := range schemas.Schemas {
-		if s.Name != name+".information_schema" {
-			schemasAPI.forceDeleteSchema(s.FullName)
-		}
-	}
-
-	return a.client.Delete(a.context, "/unity-catalog/catalogs/"+name, nil)
+	Name          string            `json:"name"`
```

Comment:
isn't the `tf:"force_new"` missing here?

Reply:
catalog name is updatable through our API but we force recreating the resource because the provider was not able to handle the change in name. Same explanation as the below's comment from Alex